assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/profiles.tf

```diff
@@ -68,6 +69,7 @@ resource "matchbox_profile" "cached-flatcar-linux-install" {
     container_linux_oem      = var.container_linux_oem
     ssh_keys                 = jsonencode(var.ssh_keys)
     install_to_smallest_disk = var.install_to_smallest_disk
+    kernel_args              = join(" ", var.kernel_args)
```

Comment:
When we use `templatefile`, we can call functions inside the template, which would be preferred way, as we keep high-level type as long as possible and we do type conversion when it's actually needed.